automates/text_reading/webapp/app/controllers/HomeController.scala

```diff
@@ -199,11 +226,12 @@ class HomeController @Inject()(cc: ControllerComponents) extends AbstractControl
     // for each block, we load the text (content) and the location of the text (page_num and block order/index on the page)
     val loader = new CosmosJsonDataLoader
     val textsAndLocations = loader.loadFile(jsonPath)
-    val texts = textsAndLocations.map(_.split("::").head)
-    val locations = textsAndLocations.map(_.split("::").tail.mkString("::")) //location = pageNum::blockIdx
+    val texts = textsAndLocations.map(_.split("<::>").slice(0,2).mkString("<::>"))
+    val locations = textsAndLocations.map(_.split("<::>").takeRight(2).mkString("<::>")) //location = pageNum::blockIdx
 
+    println("started extracting")
     // extract mentions form each text block
-    val mentions = texts.map(t => ieSystem.extractFromText(t, keepText = true, filename = Some(jsonPath)))
+    val mentions = texts.map(t => ieSystem.extractFromText(t.split("<::>").head, keepText = true, Some(t.split("<::>").last)))
```

Comment:
@maxaalexeeva this one isn't as big of a deal since testing won't involve tons of data processing, but it is still a best practice to make reusable variables for any expensive snippet of code that you would run more than once. Here you could make a variable for `t.split("<::>")` that would prevent a double computation.

(Side note: I'm coming from Python land where it's more necessary to do these types of optimizations. In Scala land your compiler may actually be doing this for you, but it is a best practice nonetheless!)